src/Integration.UnitTests/Exclusions/ExclusionSettingsFileStorageTests.cs

```diff
@@ -0,0 +1,188 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+using System;
+using System.IO;
+using System.IO.Abstractions;
+using System.Linq;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarLint.VisualStudio.Core.Exclusions;
+using SonarLint.VisualStudio.Core.Helpers;
+using SonarLint.VisualStudio.Integration.Exclusions;
+using SonarQube.Client.Models;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.Exclusions
+{
+    [TestClass]
+    public class ExclusionSettingsFileStorageTests
+    {
+        private const string NotFoundMessage = "Error loading settings for project {0}. File exclusions on the server will not be excluded in the IDE. Error: Settings File was not found";
+        private const string objectJson = "{\"Exclusions\":[\"exclusion\"],\"GlobalExclusions\":[\"globalExclusion\"],\"Inclusions\":[\"inclusion1\",\"inclusion2\"],\"GlobalInclusions\":[\"globalInclusion\"]}";
+
+        [TestMethod]
+        public void GetSettings_HaveSettings_ReadsSettings()
+        {
+            var projectKey = "projectKey";
+            var filePath = GetFilePath(projectKey);
+
+            var file = new Mock<IFile>();
+            file.Setup(f => f.ReadAllText(filePath)).Returns(objectJson);
+            file.Setup(f => f.Exists(filePath)).Returns(true);
+
+            var logger = new TestLogger();
+
+            var fileSystem = CreateFileSystem(file.Object);
+
+            var testSubject = CreateTestSubject(fileSystem, logger);
+
+            var settings = testSubject.GetSettings(projectKey);
+
+            settings.Should().NotBeNull();
+            settings.Inclusions.Count().Should().Be(2);
+            settings.Exclusions.Count().Should().Be(1);
+            settings.GlobalInclusions.Count().Should().Be(1);
+            settings.GlobalExclusions.Count().Should().Be(1);
+            logger.AssertOutputStrings(0);
+        }
+
+        [TestMethod]
+        public void GetSettings_HaveNotSettings_ReturnsNull()
```

Comment:
Cosmetics: "HaveNotSettings" can also sound as if the file exists but empty. Maybe "SettingsFileDoesNotExist" is better